src/functions/utils.php

```diff
@@ -25,3 +25,36 @@ function tribe_array_merge_recursive( array &$array1, array &$array2 ) {
 		return $merged;
 	}
 }
+
+if ( ! function_exists( 'tribe_can_plugin_run' ) ) {
+	/**
+	 * Checks if this plugin has permission to run, if not it notifies the admin
+	 *
+	 * @param string $file_path   Full file path to the base plugin file
+	 * @param string $main_class  The Main/base class for this plugin
+	 * @param string $version     The version
+	 * @param array  $classes_req Any Main class files/tribe plugins required for this to run
+	 *
+	 * @return bool Indicates if plugin should continue initialization
+	 */
+	function tribe_can_plugin_run( $file_path, $main_class, $version, $classes_req = array() ) {
```

Comment:
I'm not completely sold on the name of the function. It doesn't just return true or false according to whether dependencies are met, it also registers the new plugin if appropriate. What about `tribe_register_plugin()` instead? Additionally, we should make it clear in the docblock that it handles registration.